src/lib/components/Carousel/index.js

```diff
@@ -0,0 +1,146 @@
+  static get properties() {
+    return {
+      interval: {type: Number},
+    };
+  }
+
+  constructor() {
+    super();
+
+    /** @type {HTMLElement} */
+    this._active = undefined;
+    /** @type {HTMLElement} */
+    this._carouselTrack = undefined;
+    /** @type {number} */
+    this._index = 0;
+    /** @type {number} */
+    this.interval = 0;
+    /** @type {HTMLElement[]} */
+    this._items = [];
+    /** @type {HTMLButtonElement} */
+    this._nextButton = undefined;
+    /** @type {HTMLButtonElement} */
+    this._previousButton = undefined;
+    /** @type {NodeJS.Timeout} */
+    this._timeout = undefined;
+
+    this.next = this.next.bind(this);
+    this.previous = this.previous.bind(this);
+    this._scroll = this._scroll.bind(this);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this._carouselTrack = this.querySelector('div.carousel__track');
+
+    if (!this._carouselTrack.firstElementChild) {
+      return;
+    }
+
+    this._items = /** @type {HTMLElement[]} */ ([
+      ...this._carouselTrack.children,
+    ]);
+
+    this._items.forEach((v, i) => v.setAttribute('data-index', '' + i));
+
+    this._previousButton = this.querySelector('button[data-direction="prev"]');
+    this._previousButton.addEventListener('click', this.previous);
+    this._nextButton = this.querySelector('button[data-direction="next"]');
+    this._nextButton.addEventListener('click', this.next);
+
+    this._items.forEach((e, i) =>
+      e.addEventListener('focusin', () => this.moveSlide(i - this._index)),
+    );
+
+    this._carouselTrack.addEventListener('scroll', this._scroll);
+
+    this.moveSlide(0, false);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    if (this._previousButton)
+      this._previousButton.removeEventListener('click', this.previous);
+    if (this._nextButton)
+      this._nextButton.removeEventListener('click', this.next);
+    if (this._timeout) clearTimeout(this._timeout);
+    this._items.forEach((e) => e.parentElement.removeChild(e));
+    if (this._carouselTrack)
+      this._carouselTrack.removeEventListener('scroll', this._scroll);
+  }
+
+  moveSlide(increment, scroll = true) {
+    const elementCount = this._items.length;
+    let newIndex = this._index + increment;
+
+    if (newIndex < 0) {
+      newIndex = elementCount - 1;
+    } else if (newIndex >= elementCount) {
+      newIndex = 0;
+    }
+
+    if (newIndex === this._index) return;
+    if (this._timeout) clearTimeout(this._timeout);
+
+    this._items.forEach((e, i) => e.classList.toggle('active', i === newIndex));
+    this._index = newIndex;
+    this._active = this._items[this._index];
+    if (this.interval === 0) this._active.focus({preventScroll: true});
+    if (scroll) this._carouselTrack.scrollTo(this._active.offsetLeft, 0);
+
+    this._setTimeout();
+  }
+
+  next() {
+    this.moveSlide(1);
+  }
+
+  previous() {
+    this.moveSlide(-1);
+  }
+
+  _scroll() {
```

Comment:
Could you change this name to `_onScroll` and give it a comment? It took me a minute to work out that this is trying to, on a user's scroll event, determine the best element to make active.